### PR TITLE
RDoc-2923 RavenDB plugin for Akka.NET Persistence

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issue:**
https://issues.hibernatingrhinos.com/issue/RDoc-2923/Create-an-Akka.NET-with-RavenDB-plugin-documentation-link

### CODE REVIEW

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/queries.markdown [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+{NOTE: }
+
+
+* Akka.Persistence.Query comes with several stream-based query interfaces for querying persisted data.  
+  This interface abstracts the underlying database, allowing your application to switch persistence providers without requiring changes to the query code.
+ 
+* The RavenDB persistence plugin fully supports all Akka's query interfaces.  
+  Just include `Akka.Persistence.RavenDb.Query` in your application.
+
+* view queries in traffic watch under 'streams' todo..
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/queries.markdown:13]: right -  I'm on it now......

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown [hunk 2]
```diff
@@ -0,0 +1,178 @@
+# Events and Snapshots
+---
+
+{NOTE: }
+
+* Akka.Persistence provides two primary methods to persist actor state: **Event sourcing** and **Snapshots**.   
+  Event sourcing stores state changes as a sequence of events, while snapshots capture the actor's state at specific points for quicker recovery.
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown:7]: done

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+Persistent actors can write messages, called events, into the configured RavenDB database,  
+which serves as the events journal.
+
+**The Events collection**:  
+
+  * Each event is stored as a document in the `Events` collection in append-only mode.  
+
+  * This collection is designated by the RavenDB plugin for Akka's stored events.   
+    Although it is technically possible to store documents from other sources in this collection,  
+    it is recommended to reserve the Events collection exclusively for Akka storage.
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown:39]: done
added 'storing guidelines' section

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  * `persistentId` - The unique identifier of the actor that created the snapshot.
+  * `sequenceNr` - The sequence number indicating the position of the snapshot in the sequence of events.  
+                   Serves as a unique, gap-less identifier that helps maintain the correct order and consistency of the actor's state.
+
+**Replaying snapshots**:  
+
+  * Replaying snapshots significantly accelerates recovery, reduces network transmission,  
+    and lowers both actor event replay time and CPU usage.
+
+  * After the snapshot is retrieved, events can be deleted from the events journal to free up space.
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown:90]: done

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/queries.markdown [hunk 5]
```diff
@@ -0,0 +1,202 @@
+# Queries
+---
+
+{NOTE: }
+
+
+* Akka.Persistence.Query comes with several stream-based query interfaces for querying persisted data.  
+  This interface abstracts the underlying database, allowing your application to switch persistence providers without requiring changes to the query code.
+ 
+* The RavenDB persistence plugin fully supports all Akka's query interfaces.
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/queries.markdown:10]: done

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+  * Events must be applied (replayed) in the exact order they were generated,  
+    so consistency is crucial to ensure that no events are missed or processed out of order.
+
+**Cluster-wide transactions**:  
+
+  * RavenDB is a distributed database, allowing writes, reads, and queries to target different nodes across the cluster.  
+
+  * To prioritize consistency over availability, the RavenDB plugin uses a [cluster-wide transaction](../../server/clustering/cluster-transactions) for storing events and snapshot documents. 
+    This ensures that persisted data is consistently applied across all database instances in the cluster, fulfilling Akka's requirement for no gaps in event sequence numbers at any point in time, across all nodes.
```
Comment (karmeli87) [inline Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown:121]: it is not about the gaps, we require the cluster wide tx in order to restoring to the latest state and avoid conflicts

File: Documentation/6.1/Samples/csharp/Raven.Documentation.Samples/Integrations/AkkaPersistence/hocon-configuration.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
+/*
+#region hocon_config 
+<akka>
+  <hocon>
+  <![CDATA[
```
Comment (karmeli87) [inline Documentation/6.1/Samples/csharp/Raven.Documentation.Samples/Integrations/AkkaPersistence/hocon-configuration.cs:5]: hmmm not familiar with this such style..

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/integrating-with-akka-persistence.markdown [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+  * Persist and recover Events to/from a **Journal store**. Learn more in [Events](../../integrations/akka.net-persistence/events-and-snapshots#storing-events).
+  * Persist and recover Snapshots to/from a **Snapshot store**. Learn more in [Snapshots](../../integrations/akka.net-persistence/events-and-snapshots#storing-snapshots). 
+  * Query the stored events. Learn more in [Queries](../../integrations/akka.net-persistence/queries).
+
+{PANEL/}
+
+{PANEL: Installing the RavenDB persistence plugin}
+
+Integrate RavenDB with Akka.Persistence using one of the two available NuGet packages:
```
Comment (karmeli87) [inline Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/integrating-with-akka-persistence.markdown:67]: What I miss here is a command that I can run with a copy-paste to install this in my solution

**Review by karmeli87 — COMMENTED** [review]
Very clear and comprehensive, well done! :-)
I think we can add links to the 2 articles we have
https://ravendb.net/articles/using-ravendb-persistence-in-an-akka-net-application
https://ravendb.net/articles/notes-from-integrating-ravendb-with-akka-net-persistence

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown (hunk 6 above)
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/events-and-snapshots.markdown:121]: Modified to:
```
This ensures that persisted data is consistently applied across all database instances in the cluster, 
preventing conflicts and guaranteeing that restoring to the latest state reflects the correct event sequence, 
as required by Akka.
```

File: Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/integrating-with-akka-persistence.markdown (hunk 8 above)
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/integrations/akka.net-persistence/integrating-with-akka-persistence.markdown:67]: done

**Pull request merged**